crates/staged-sync/src/utils/init.rs

```diff
@@ -21,23 +21,32 @@ pub fn init_db<P: AsRef<Path>>(path: P) -> eyre::Result<Env<WriteMap>> {
     Ok(db)
 }
 
+/// Database initialization error type.
+#[derive(Debug, thiserror::Error, PartialEq, Eq, Clone)]
+enum InitDatabaseError {
+    /// Attempted to reinitialize database with inconsistent genesis block
+    #[error("Genesis hash mismatch: expected {expected}, got {actual}")]
+    GenesisHashMismatch { expected: H256, actual: H256 },
+}
+
 /// Write the genesis block if it has not already been written
-///
-/// # Panics
-///
-/// Panics if the database has already been initialized with a different genesis block.
 #[allow(clippy::field_reassign_with_default)]
-pub fn init_genesis<DB: Database>(db: Arc<DB>, chain: ChainSpec) -> Result<H256, reth_db::Error> {
+pub fn init_genesis<DB: Database>(db: Arc<DB>, chain: ChainSpec) -> eyre::Result<H256> {
```

Comment:
this should return InitDatabaseError so we have dedicated error types in all lib crates and not eyre.

Reply:
No worries, will make this switch.